src/utils/getUrlByRequestOptions.ts

```diff
@@ -75,8 +75,20 @@ function getPortByRequestOptions(
   return undefined
 }
 
-function getHostByRequestOptions(host?: string | null): string {
-  return host || DEFAULT_HOST
+function getHostByRequestOptions(options: ResolvedRequestOptions): string {
+  if (options.host) {
+    return options.host
+  }
+
+  if (options.hostname) {
+    if (options.port) {
+      return `${options.hostname}:${options.port}`
```

Comment:
Just to be sure, would this work for IPv6 hosts as well?

Reply:
AFAIK:
If set, the hostname should be: `[ipv6]`
If set, the port should be: `8080`

In conjunction: `[ipv6]:8080` which is a valid `host

Reply:
Thanks for clarifying!